marconi-chain-index/src/Marconi/ChainIndex/Indexers.hs

```diff
@@ -723,10 +728,12 @@ runIndexers
           "Stopping indexing. Waiting for indexers to finish their work (timeout after "
             <> Text.pack (show secondsBeforeTimeout)
             <> "s) ..."
-        void $
+        res <-
           timeout (secondsBeforeTimeout * 1_000_000) $
             waitQSemN (coordinator ^. barrier) (coordinator ^. indexerCount)
-        logInfo stdoutTrace "Done!"
+        case res of
+          Just _ -> logInfo stdoutTrace "Done!"
+          Nothing -> throwIO (Timeout @Void "Timed out.") -- TODO: What useful info can go here?
```

Comment:
It's hard to tell, as we don't know which indexer did timeout here.
It's likely that in the new indexer implementation, we'll try to move the timeout in each worker, to attach the worker name to the error message.